src/main/scala/strawman/collection/immutable/Map.scala

```diff
@@ -27,6 +27,18 @@ trait MapOps[K, +V, +CC[X, +Y] <: Map[X, Y] with MapOps[X, Y, CC, _], +C <: Map[
   /** Alias for `remove` */
   @`inline` final def - (key: K): C = remove(key)
 
+  /** Creates a new $coll from this $coll by removing all elements of another
+    *  collection.
+    *
+    *  @param keys   the collection containing the removed elements.
+    *  @return a new $coll that contains all elements of the current $coll
+    *  except one less occurrence of each of the elements of `elems`.
+    */
+  def removeAll(keys: IterableOnce[K]): C = fromSpecificIterable(keys.iterator().foldLeft(coll)(_ - _))
```

Comment:
Dotty reports a type error here:

~~~
def removeAll(keys: IterableOnce[K]): C = fromSpecificIterable(keys.iterator().foldLeft(coll)(_ - _))
                                                                                              ^^^^^
                                                    found:    C
                                                    required: CC[K, V]
~~~

But I think _he_ is wrong.
Indeed, the type of `coll` is `CC[K, V]`, which is upper bounded by `MapOps[K, V, CC, CC[K, V]]` (note that the last parameter (whose formal name is `C`) is instantiated to `CC[K, V]`). Then, the `-` operator returns a `C`, which is a `CC[K, V]`.